addons/mail/static/src/components/discuss/tests/discuss_inbox_tests.js

```diff
@@ -411,6 +411,100 @@ QUnit.test('"reply to" composer should send message if message replied to is not
     assert.verifySteps(['message_post']);
 });
 
+QUnit.test('show subject of message in inbox', async function (assert) {
+    assert.expect(3);
+
+    this.data['mail.message'].records = [{
+        author_id: [7, "Demo"],
+        body: "<p>Test</p>",
+        date: "2019-04-20 11:00:00",
+        id: 100,
+        is_discussion: false,
+        is_notification: false,
+        message_type: 'comment',
+        needaction: true,
+        needaction_partner_ids: [3],
+        model: 'mail.channel',
+        res_id: 20,
+        subject: "Salutations, voyageur",
+    }];
+
+    await this.start({
+        env: {
+            session: {
+                name: 'Admin',
+                partner_display_name: 'Your Company, Admin',
+                partner_id: 3,
+                uid: 2,
+            },
+        },
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "should display a single message"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message_subject',
+        "should display subject of the message"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_Message_subject').textContent,
+        "Subject: Salutations, voyageur",
+        "Subject of the message should be 'Salutations, voyageur'"
+    )
+});
+
+QUnit.test('show subject of message in history', async function (assert) {
+    assert.expect(3);
+
+    this.data['mail.message'].records = [{
+        author_id: [7, "Demo"],
+        body: "<p>Test</p>",
+        date: "2019-04-20 11:00:00",
+        id: 100,
+        message_type: 'comment',
+        history_partner_ids: [3],
+        model: 'project.task',
+        record_name: "Refactoring",
+        res_id: 20,
+        subject: "Salutations, voyageur",
+    }];
+
+    await this.start({
+        discuss: {
+            params: {
+                default_active_id: 'mail.box_history',
+            },
+        },
+        env: {
+            session: {
+                name: 'Admin',
+                partner_display_name: 'Your Company, Admin',
+                partner_id: 3,
+                uid: 2,
+            },
+        },
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "should display a single message"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message_subject',
+        "should display subject of the message"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_Message_subject').textContent,
+        "Subject: Salutations, voyageur",
+        "Subject of the message should be 'Salutations, voyageur'"
+    )
+});
+
+
```

Comment:
Only one new line at the end.

Shouldn't we also add a test to ensure subject is specially not shown in other cases?

Reply:
Where would you place that test ? in discuss_tests.js and selecting a random channel ?

Reply:
Would not add such test(s). It's annoying to test what we don't want to have, especially when the amount of possibilities are significant (in channels, mailing channels, chatter, on chat window, etc...).

Besides not displaying subjects is debatable, and the main reason to not display is about UI and noise (due to misuse of "subject" when it's the same as the record name? Up to debate...).

Reply:
Ok to ignore this suggestion for now, we have more important tasks.

Reply:
ok resolving this conv then